src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -579,8 +590,11 @@ castString = veryUnsafeCoerceSqlExprValue
 
 -- | Execute a subquery @SELECT@ in an SqlExpression.  Returns a
 -- list of values.
-subList_select :: PersistField a => SqlQuery (SqlExpr (Value a)) -> SqlExpr (ValueList a)
-subList_select         = EList . sub_select
+subList_select
+  :: PersistField a
+  => SqlQuery (SqlExpr (Value a))
+  -> SqlExpr (ValueList a)
+subList_select         = EList . unsafePromiseNotNull . sub_select
```

Comment:
I *think* this should do a `NULL -> []` transformation but I am not sure. Will need to write a test against it.